src/services/cw721/cw721-activity-update-owner.service.ts

```diff
@@ -0,0 +1,125 @@
+import { Service } from '@ourparentcenter/moleculer-decorators-extended';
+import _ from 'lodash';
+import { ServiceBroker } from 'moleculer';
+import config from '../../../config.json' assert { type: 'json' };
+import BullableService, { QueueHandler } from '../../base/bullable.service';
+import { Config } from '../../common';
+import { BULL_JOB_NAME, SERVICE } from '../../common/constant';
+import CW721Activity from '../../models/cw721_tx';
+import { CW721_ACTION } from './cw721.service';
+
+const { NODE_ENV } = Config;
+
+@Service({
+  name: SERVICE.V1.Cw721ActivityUpdateOwnerService.key,
+  version: 1,
+})
+export default class Cw721ActivityUpdateOwnerService extends BullableService {
+  public constructor(public broker: ServiceBroker) {
+    super(broker);
+  }
+
+  @QueueHandler({
+    queueName: BULL_JOB_NAME.UPDATE_CW721_ACTIVITY_OWNER_BY_TOKEN,
+    jobName: BULL_JOB_NAME.UPDATE_CW721_ACTIVITY_OWNER_BY_TOKEN,
+  })
+  async updateCw721ActOwnerByToken(_payload: {
```

Comment:
hiện tại trên xstaxy cũng chỉ có 25k activity, load hết ra mà sửa trong 1 migration
sau đó thì gán luôn lúc có activity mới

Reply:
Trước kia em đã có nghĩ giải quyết theo hướng chạy migration update data cũ và sửa logic code. Nhưng việc sửa logic code sẽ gặp những vấn đề là: 

- code phình to
- logic tương đối phức tạp
- khó debug
- có nhiều trường hợp khi một phần data nằm trong range [startBlock, endBlock] khi mình bắt được và một phần data nằm trong DB

Em muốn thống nhất là data khi trong DB rồi thì mình bốc ra xử lý những cái chưa update owner